lib/nanoc/data_sources/filesystem.rb

```diff
@@ -58,10 +58,12 @@ def load_objects(dir_name, kind, klass)
           if is_binary && klass == Nanoc::Int::Item
             meta                = (meta_filename && YAML.load_file(meta_filename)) || {}
             content_or_filename = content_filename
+            checksum = nil
           elsif is_binary && klass == Nanoc::Int::Layout
             raise "The layout file '#{content_filename}' is a binary file, but layouts can only be textual"
           else
-            meta, content_or_filename = parse(content_filename, meta_filename, kind)
+            meta, content_or_filename, meta_raw = parse(content_filename, meta_filename, kind)
+            checksum = "meta=#{meta_raw},content=#{content_or_filename}"
```

Comment:
The only consumer of the checksum is `Nanoc::Int::Checksummer.calc`. Does it really need to be sent through the checksummer a second time?